Detectors/EMCAL/simulation/src/Detector.cxx

```diff
@@ -165,50 +167,81 @@ Bool_t Detector::ProcessHits(FairVolume* v)
   LOG(DEBUG3) << "Name of the supermodule type " << fMC->CurrentVolOffName(4) << ", Module name "
               << fMC->CurrentVolOffName(3);
 
-  Int_t partID = fMC->GetStack()->GetCurrentTrackNumber(),
-        parent = fMC->GetStack()->GetCurrentTrack()->GetMother(0),
-        detID = geom->GetAbsCellId(offset + copySmod - 1, copyMod - 1, copyPhi - 1, copyEta - 1);
+  // porting from aliroot: circumvention of the index problem. GetAbsCellID cannot be used
+  // Int_t detID = geom->GetAbsCellId(offset + copySmod - 1, copyMod - 1, copyPhi - 1, copyEta - 1);
+  //
+  // Swap A side in Phi and C side in Eta due to wrong indexing.
+  //
+  Int_t smNumber = offset + copySmod - 1, smTypeID = 1;
+  auto [iphi, ieta] = geom->GetCellPhiEtaIndexInSModule(smNumber, copyMod - 1, copyPhi - 1, copyEta - 1);
+  //iphi = std::get<0>(posetaphi);
+  //ieta = std::get<1>(posetaphi);
+  if (smNumber % 2 == 0) {
+    if (strcmp(fMC->CurrentVolOffName(4), "DCSM") == 0)
+      smTypeID = 3; //DCal supermodule. previous design/idea
+    else
+      smTypeID = 2;
+    ieta = ((geom->GetCentersOfCellsEtaDir()).size() * 2 / smTypeID - 1) - ieta; // 47/31-ieta, revert the ordering on A side in order to keep convention.
+  } else {
+    if (strcmp(fMC->CurrentVolOffName(4), "SM10") == 0)
+      smTypeID = 2; //half supermodule. previous design/idea
+    if (strcmp(fMC->CurrentVolOffName(4), "SM3rd") == 0)
+      smTypeID = 3; //one third (installed in 2012) supermodule
+    if (strcmp(fMC->CurrentVolOffName(4), "DCEXT") == 0)
+      smTypeID = 3;                                                          //one third (installed in 2012) supermodule
+    iphi = ((geom->GetCentersOfCellsPhiDir()).size() / smTypeID - 1) - iphi; // 23/7-iphi, revert the ordering on C side in order to keep convention.
```

Comment:
here and above it would be preferred and faster to do the checks based on cached volume IDs instead of repeated string comparisons. Could you schedule such optimization for a separate PR maybe?

Reply:
For this part I could make use of the supermodule type determined above using the enum class and a switch. In general replacing with cached volume IDs would need a second PR.

Reply:
fine for me. Just wanted to raise the idea. Could you trace this task in a ticket and follow up separately?